braidpool-primitives/src/bead/mod.rs

```diff
@@ -1,34 +1,110 @@
 #![allow(unused)]
+use bitcoin::consensus::Error;
+use bitcoin::io::{self, BufRead, Write};
+use bitcoin::p2p::address::AddrV2;
 // Standard Imports
 use ::serde::{Deserialize, Serialize};
 use std::cell::RefCell;
 use std::collections::{HashMap, HashSet};
-use std::net::SocketAddr;
 
 // Bitcoin primitives
 use crate::braid::Braid;
 use bitcoin::absolute::Time;
+use bitcoin::consensus::encode::Decodable;
+use bitcoin::consensus::encode::Encodable;
+use bitcoin::consensus::encode::serialize;
 use bitcoin::ecdsa::Signature;
 use bitcoin::p2p::Address as P2P_Address;
 use bitcoin::secp256k1::PublicKey;
 use bitcoin::transaction::TransactionExt;
 use bitcoin::{Address, BlockHash};
 use bitcoin::{BlockHeader, Transaction};
+use core::str::FromStr;
 // Custom Imports
 use crate::utils::{BeadHash, BeadLoadError, Children, Parents};
-#[derive(Clone, Debug, Serialize)]
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Default)]
+pub struct TimeVec(pub Vec<Time>);
+
+impl Encodable for TimeVec {
+    fn consensus_encode<W: Write + ?Sized>(&self, w: &mut W) -> Result<usize, io::Error> {
+        let mut len = 0;
+        // Encode the length for deterministic encoding
+        len += (self.0.len() as u64).consensus_encode(w)?;
+        for time in &self.0 {
+            len += time.to_consensus_u32().consensus_encode(w)?;
+        }
+        Ok(len)
+    }
+}
+
+impl Decodable for TimeVec {
+    fn consensus_decode<R: BufRead + ?Sized>(r: &mut R) -> Result<Self, Error> {
+        let len = u64::consensus_decode(r)?;
+        let mut vec = Vec::with_capacity(len as usize);
+        for _ in 0..len {
+            let time_u32 = u32::consensus_decode(r)?;
+            let time = Time::from_consensus(time_u32).unwrap();
+            vec.push(time);
+        }
+        Ok(TimeVec(vec))
+    }
+}
+
+#[derive(Clone, Debug)]
 
 pub struct CommittedMetadata {
     // Committed Braidpool Metadata,
     pub transaction_cnt: u32,
```

Comment:
Is a count needed here? Doesn't the serialization of `Vec<Transaction>` include a count?